### PR TITLE
KHR_materials_transmission: Fix transmission helper to generate a linear space opaque texture

### LINKED ISSUE #10313: KHR_materials_transmission: generate the opaque texture in linear space and without post processing
See https://forum.babylonjs.com/t/nodematerials-env-maps-tonemapping-and-exposure/20223/21

Currently, the opaque texture is generated in gamma space and with the image processing currently in effect on the scene. That means the display is incorrect when image processing is enabled at the scene level because the image processing is applied twice, when generating the texture and when displaying the meshes that use this texture as a refraction texture.

We should instead always generate the opaque texture in linear space and without any image processing applied. This way it will work in all cases, the image processing being applied at the material or at the scene level.

### PR DESCRIPTION
**Pull request by Popov72**
See https://forum.babylonjs.com/t/nodematerials-env-maps-tonemapping-and-exposure/20223/22

Fix #10313

### CODE REVIEW

File: loaders/src/glTF/2.0/Extensions/KHR_materials_transmission.ts [hunk 1]
```diff
@@ -219,12 +226,27 @@ class TransmissionHelper {
...
+
+        this._opaqueRenderTarget.onBeforeBindObservable.add(() => {
+            sceneImageProcessingapplyByPostProcess = this._scene.imageProcessingConfiguration.applyByPostProcess;
+            if (!this._options.clearColor) {
+                this._scene.clearColor.toLinearSpaceToRef(this._opaqueRenderTarget!.clearColor);
+            } else {
+                this._opaqueRenderTarget!.clearColor = this._options.clearColor;
+            }
+            this._scene.imageProcessingConfiguration.applyByPostProcess = true;
+        });
```
Comment (bghgary) [inline loaders/src/glTF/2.0/Extensions/KHR_materials_transmission.ts:245]: nit:
```suggestion
        this._opaqueRenderTarget.onBeforeBindObservable.add((opaqueRenderTarget) => {
            sceneImageProcessingapplyByPostProcess = this._scene.imageProcessingConfiguration.applyByPostProcess;
            if (!this._options.clearColor) {
                this._scene.clearColor.toLinearSpaceToRef(opaqueRenderTarget.clearColor);
            } else {
                opaqueRenderTarget.clearColor.copyFrom(this._options.clearColor);
            }
            this._scene.imageProcessingConfiguration.applyByPostProcess = true;
        });
```

**Review by bghgary — APPROVED** [review]

File: loaders/src/glTF/2.0/Extensions/KHR_materials_transmission.ts [hunk 2]
```diff
@@ -219,12 +226,27 @@ class TransmissionHelper {
         this._opaqueRenderTarget = new RenderTargetTexture("opaqueSceneTexture", this._options.renderSize, this._scene, true, undefined, this._options.renderTargetTextureType);
         this._opaqueRenderTarget.ignoreCameraViewport = true;
         this._opaqueRenderTarget.renderList = this._opaqueMeshesCache;
-        // this._opaqueRenderTarget.clearColor = new Color4(0.0, 0.0, 0.0, 0.0);
-        this._opaqueRenderTarget.gammaSpace = true;
+        this._opaqueRenderTarget.clearColor = this._options.clearColor ?? this._scene.clearColor.clone();
```
[the remark this answers is not in the document]
Comment (Popov72) [reply loaders/src/glTF/2.0/Extensions/KHR_materials_transmission.ts:229]: `_opaqueRenderTarget.clearColor` can be undefined (it's the default value, meaning RTT will use the `scene.clearColor` instead).

[the remark this answers is not in the document]
Comment (bghgary) [reply loaders/src/glTF/2.0/Extensions/KHR_materials_transmission.ts:229]: I see that. Not sure what you mean by this. My comment is about the copying vs. assignment.

[the remark this answers is not in the document]
Comment (Popov72) [reply loaders/src/glTF/2.0/Extensions/KHR_materials_transmission.ts]: ok, fixed after offline discussion

**Pull request merged**